tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3662,59 +3815,121 @@ <h2>Create a distance feature</h2>
       },
 
       makeBarChart: function(featureName, data){
-        // categorical data:
-        // {"1": [{"step": ["Divorced"], "scalar": 0.1533520370721817},
-        //        {"step": ["Married-AF-spouse"], "scalar": 0.153403326869010}]}
+        // array of categorical data objects (keys are labels):
+        // {"1": [{"step": ["Divorced"], "scalar": 0.1533520370721},
+        //    {"step": ["Married-AF-spouse"], "scalar": 0.153403326869}]},
+        // one object per model
 
-        function convertPoint(point){
+        function convertPoint(point, origKey){
           // point: {"step": ["Divorced"], "scalar": 0.1533520370721817}
           return {
-            x: point.step,
+            x: (point.step ==
+              origKey) ? point.step + ' - original value' : point.step,
             y: point.scalar
           };
         };
 
-        // Make a series for the current value of the feature to show in the
-        // plot.
+        // Overwrite the key for original value so that it reads under the bar
+        let origKey = '';
+        const firstModelValues = data[0];
         if (!this.globalPdPlots) {
           const origValEntry =
-            this.createPdEntryForCurrentValue(featureName, data);
+            this.createPdEntryForCurrentValue(featureName, firstModelValues, 0);
           if (origValEntry) {
-            let indexOfOrigValInData = -1;
-            const origKey = _.keys(data)[0];
-            const origData = data[origKey];
-            for (let i = 0; i < origData.length; i++) {
-              if (origData[i].step == origValEntry.step) {
-                indexOfOrigValInData = i;
-                break;
-              }
-            }
-            if (indexOfOrigValInData != -1) {
-              origData.splice(indexOfOrigValInData, 1);
-            }
-            data[origKey + ' - original value'] = [origValEntry];
+            origKey = origValEntry.step;
           }
         }
 
         // Convert the data into the format expected by vz-bar-chart.
-        const mapped = _.mapValues(data, function(pointsForSeries){
-          return _.map(pointsForSeries, point => convertPoint(point));
+        // Stack models and labels so that each (model,label) is a separate
+        // series
+        self = this;
+        const mapped = {};
+        _.forEach(data, function(modelValues, modelInd){
```

Comment:
i know the code was like this before, but if you remove function() {} and replace with () => {} here and below, then you don't need the self=this, can just use this.  also if you were to use "self", should be "const self"

Reply:
done.